evaluate/task/task-transpile.go

```diff
@@ -170,22 +162,38 @@ func (t *TaskTranspile) Run(ctx evaltask.Context) (repositoryAssessment map[eval
 	return repositoryAssessment, problems, nil
 }
 
-// unpackTranspilerPackage checks if the testdata repository for the transpilation task is well-formed and returns the path to the implementation file and also the path to the file that holds the stub.
-func (t *TaskTranspile) unpackTranspilerPackage(ctx evaltask.Context, fileLogger *log.Logger, originLanguage language.Language, packagePath string) (originFilePath string, stubFilePath string, err error) {
+// unpackTranspilerPackage returns the path to the implementation file and also the path to the file that holds the stub.
+func (t *TaskTranspile) unpackTranspilerPackage(ctx evaltask.Context, fileLogger *log.Logger, packagePath string) (originFilePathsWithLanguage map[string]language.Language, stubFilePath string, err error) {
+	originFilePathsWithLanguage = map[string]language.Language{}
 	packagePathAbsolute := filepath.Join(ctx.Repository.DataPath(), packagePath)
 
-	files, err := originLanguage.Files(fileLogger, filepath.Join(packagePathAbsolute, "implementation"))
+	transpilationDirectories, err := os.ReadDir(filepath.Join(packagePathAbsolute, "implementation"))
 	if err != nil {
-		return "", "", pkgerrors.WithStack(err)
+		return nil, "", pkgerrors.WithStack(err)
+	}
+	for _, transpilationDirectory := range transpilationDirectories {
+		transpilationDirectoryPath := filepath.Join(packagePathAbsolute, "implementation", transpilationDirectory.Name())
+		languageID := strings.Split(transpilationDirectory.Name(), "-")[1]
+		var originLanguage language.Language
+		for _, l := range language.Languages {
+			if languageID == l.ID() {
+				originLanguage = l
+			}
+		}
+		files, err := originLanguage.Files(fileLogger, transpilationDirectoryPath)
+		if err != nil {
+			return nil, "", err
+		}
+		newOriginLanguage := originLanguage
```

Comment:
Why the re-pointer. Since the language is already a pointer to begin with, don't think you need it.